tests/test_quickbooks_bookmarks.py

```diff
@@ -42,33 +40,89 @@ def test_run(self):
 
         # Verify tap and target exit codes
         exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
         menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
 
-        # Verify the bookmark is correct, LastUpdatedTime assumes no other Account records are added to Sandbox
-        actual_state = menagerie.get_state(conn_id)
-        expected_state = {'bookmarks': {'accounts': {'LastUpdatedTime': '2020-08-25T13:17:37-07:00'}}}
-        self.assertEqual(actual_state, expected_state)
+        # UPDATE STATE BETWEEN SYNCS
+        new_state = dict()
+        new_state['bookmarks'] = {key: {'LastUpdatedTime': value} for key, value in self.simulated_states_by_stream().items()}
+        menagerie.set_state(conn_id, new_state)
 
-        # Verify actual rows were synced
-        sync_record_count = runner.examine_target_output_file(
+        # SYNC 2
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_record_count = runner.examine_target_output_file(
             self, conn_id, self.expected_streams(), self.expected_primary_keys())
+        second_sync_bookmark = menagerie.get_state(conn_id)
+
+        # Verify tap and target exit codes
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
 
-        # Examine target output
+        # Test by stream
         for stream in self.expected_streams():
             with self.subTest(stream=stream):
-                # Each stream should have 1 or more records returned
-                self.assertGreaterEqual(sync_record_count[stream], 1)
-
-        # Update state and sync again
-        new_state = {'bookmarks': {'accounts': {'LastUpdatedTime': '2020-08-25T13:17:36-07:00'}}}
-        menagerie.set_state(conn_id, new_state, version=1)
-
-        sync_job_name = runner.run_sync_mode(self, conn_id)
-
-        # Verify only 1 record synced with the new state
-        sync_record_count = runner.examine_target_output_file(
-            self, conn_id, self.expected_streams(), self.expected_primary_keys())
-        self.assertEqual(sync_record_count, {'accounts': 1})
+                # record counts
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+
+                # record messages
+                first_sync_messages = first_sync_records.get(stream, {'messages': []}).get('messages')
+                second_sync_messages = second_sync_records.get(stream, {'messages': []}).get('messages')
+
+                # replication key is an object (MetaData.LastUpdatedTime) in sync records
+                # but just the sub level pk is used in setting bookmarks
+                top_level_rk = 'MetaData'
+                sub_level_rk = 'LastUpdatedTime'
```

Comment:
Maybe for a subsequent pr, I personally don't like `rk` or shortened variable names like this. Please just call it top_level_replication_key.